lua/dap/repl.lua

```diff
@@ -31,7 +47,8 @@ local function new_buf()
   api.nvim_buf_set_keymap(buf, 'n', 'o', "<Cmd>lua require('dap.ui').trigger_actions()<CR>", {})
   api.nvim_buf_set_keymap(buf, 'i', '<up>', "<Cmd>lua require('dap.repl').on_up()<CR>", {})
   api.nvim_buf_set_keymap(buf, 'i', '<down>', "<Cmd>lua require('dap.repl').on_down()<CR>", {})
-  vim.fn.prompt_setprompt(buf, 'dap> ')
+  vim.fn.setbufvar(buf, PROMPT_STATE, REGULAR_PROMPT)
+  vim.fn.prompt_setprompt(buf, get_prompt(buf))
   vim.fn.prompt_setcallback(buf, execute)
```

Comment:
We could consider adding `<C-CR>` and `<S-CR>` mappings where ctrl-enter is for "always submit" and shift-enter is for "always start newline for multiline expression". A regular "<CR>" can then be the smart variant which uses the adapters multiline detection.

(Although this will require configuration for some terminals: https://stackoverflow.com/a/42461580)